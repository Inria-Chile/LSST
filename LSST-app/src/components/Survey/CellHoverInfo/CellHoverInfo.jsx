import React, { PureComponent } from 'react'
import './CellHoverInfo.css';
import { filterColors, decreaseBrightness } from "../../Utils/Utils"

class CellHoverInfo extends PureComponent {
    
    render() {
        return (
            <div className="hover-div">
                <div>FieldID: {this.props.selectedField && this.props.selectedField.fieldID ? this.props.selectedField.fieldID: ''}</div>
                <div>Timestamp: {this.props.selectedField && this.props.selectedField.expDate ? this.props.selectedField.expDate: ''}</div>
                <div>
                    Filter: {" "}
                    <div className="hover-filter" style={{
                            backgroundColor: filterColors[this.props.selectedField.filterName ] ? filterColors[this.props.selectedField.filterName ]
                                : "#000000", border: decreaseBrightness(filterColors[this.props.selectedField.filterName], 1.3) + ' solid 2px'
                        }}>
                        {
                            this.props.selectedField.filterName ? this.props.selectedField.filterName : ''
                        }
                    </div>
                </div>
            </div>
        );
    }
}


export default CellHoverInfo;