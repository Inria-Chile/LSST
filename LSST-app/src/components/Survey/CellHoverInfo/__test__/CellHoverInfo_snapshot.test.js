import React from 'react';
import renderer from 'react-test-renderer';
import shallowRenderer from 'react-test-renderer/shallow';
import CellHoverInfo from '../CellHoverInfo';



/*---- Snapshot test ----*/
describe('CellHoverInfo_Test',function(){
    it('renders correctly with filtername setted',()=> {
        const selectedField = {filterName : 'x'};
        const renderer = new shallowRenderer();
        const tree = renderer.render(
          <CellHoverInfo selectedField={selectedField}/>);
          expect(tree).toMatchSnapshot();
      });

      it('renders correctly with fieldId and expDate setted',()=>{
        const selectedField = {
            filterName : 'x',
            fieldID : 10,
            expDate : "2018-01-04T19:44:10.611Z",
        };
        const renderer = new shallowRenderer();
        const tree = renderer.render(
            <CellHoverInfo selectedField={selectedField}/>);
        expect(tree).toMatchSnapshot();

      });    
});

