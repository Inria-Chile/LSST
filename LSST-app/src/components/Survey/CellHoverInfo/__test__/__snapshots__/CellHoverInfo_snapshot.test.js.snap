// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CellHoverInfo_Test renders correctly with fieldId and expDate setted 1`] = `
<div
  className="hover-div"
>
  <div>
    FieldID: 
    10
  </div>
  <div>
    Timestamp: 
    2018-01-04T19:44:10.611Z
  </div>
  <div>
    Filter: 
     
    <FilterIndicator
      filterName="x"
    />
  </div>
</div>
`;

exports[`CellHoverInfo_Test renders correctly with filtername setted 1`] = `
<div
  className="hover-div"
>
  <div>
    FieldID: 
    
  </div>
  <div>
    Timestamp: 
    
  </div>
  <div>
    Filter: 
     
    <FilterIndicator
      filterName="x"
    />
  </div>
</div>
`;
